src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp

```diff
@@ -347,6 +359,20 @@ SelectPartsDecision MergeTreeDataMergerMutator::selectPartsToMerge(
 
         if (parts_to_merge.empty())
         {
+            if (data_settings->min_age_to_force_merge_on_partition_only && data_settings->min_age_to_force_merge_seconds)
+            {
+                auto best_partition_it = std::max_element(
+                    partitions_info.begin(),
+                    partitions_info.end(),
+                    [](const auto & e1, const auto & e2) { return e1.second.min_age > e2.second.min_age; });
```

Comment:
```suggestion
                    [](const auto & e1, const auto & e2) { return e1.second.min_age < e2.second.min_age; });
```
Not sure, but looks like you want to find the oldest partition?

Reply:
you're correct, nice catch!